core/src/main/java/hudson/security/HudsonPrivateSecurityRealm.java

```diff
@@ -497,13 +502,41 @@ public boolean isMailerPluginPresent() {
 
     /**
      * Creates a new user account by registering a password to the user.
+     *
+     * Supports bcrypt hashed passwords if the password begins with the characters
+     * <code>#jbcrypt:</code>
      */
     public User createAccount(String userName, String password) throws IOException {
         User user = User.getById(userName, true);
         user.addProperty(Details.fromPlainPassword(password));
         return user;
     }
 
+    /**
+     * Creates a new user account by registering a JBCrypt Hashed password with the user.
+     *
+     * @param userName The user's name
+     * @param hashedPassword A hashed password, must begin with <code>#jbcrypt:</code>
+     */
+    public User createAccountWithHashedPassword(String userName, String hashedPassword) throws IOException {
+        if (!hashedPassword.startsWith(JBCRYPT_HEADER)) {
+            throw new IllegalArgumentException("Hashed passwords must start with: "+ JBCRYPT_HEADER);
+        }
+        User user = User.getById(userName, true);
+        user.addProperty(Details.fromHashedPassword(hashedPassword));
+        return user;
+    }
+
+    /**
+     * Returns true if the supplied password starts with a prefix indicating it is already hashed.
+     */
+    public static boolean isPasswordHashed(String password) {
```

Comment:
Maybe it would make more sense to move this into the `PASSWORD_ENCODER`. Seems like that would better isolate things.

What is this method needed for? Other than possibly as a replacement in the earlier check.

Reply:
Ya, that seems a lot more reasonable given the other changes I've made. Thanks for the suggestion.